nowcasting_dataset/dataset/validate.py

```diff
@@ -0,0 +1,342 @@
+        x["nwp_y_coords"], _ = torch.sort(
+            torch.randn(self.batch_size, self.nwp_image_size_pixels), descending=True
+        )
+
+        # add sorted (fake) time series
+        x["sat_datetime_index"], _ = torch.sort(torch.randn(self.batch_size, self.seq_length_5))
+        x["nwp_target_time"], _ = torch.sort(torch.randn(self.batch_size, self.seq_length_5))
+        x["gsp_datetime_index"], _ = torch.sort(torch.randn(self.batch_size, self.seq_length_30))
+
+        x["x_meters_center"], _ = torch.sort(torch.randn(self.batch_size))
+        x["y_meters_center"], _ = torch.sort(torch.randn(self.batch_size))
+
+        # clip yield values from 0 to 1
+        x["pv_yield"] = torch.clip(x["pv_yield"], min=0, max=1)
+        x["gsp_yield"] = torch.clip(x["gsp_yield"], min=0, max=1)
+
+        return x
+
+
+def validate_example(
+    data: Example,
+    seq_len_30_minutes: int,
+    seq_len_5_minutes: int,
+    sat_image_size: int = 64,
+    n_sat_channels: int = 1,
+    nwp_image_size: int = 0,
+    n_nwp_channels: int = 1,
+    n_pv_systems_per_example: int = DEFAULT_N_PV_SYSTEMS_PER_EXAMPLE,
+    n_gsp_per_example: int = DEFAULT_N_GSP_PER_EXAMPLE,
+    batch: bool = False,
+):
+    """
+    Validate the size and shape of the data
+    Args:
+        data: Typed dictionary of the data
+        seq_len_30_minutes: the length of the sequence for 30 minutely data
+        seq_len_5_minutes: the length of the sequence for 5 minutely data
+        sat_image_size: the satellite image size
+        n_sat_channels: the number of satellite channgles
+        nwp_image_size: the nwp image size
+        n_nwp_channels: the number of nwp channels
+        n_pv_systems_per_example: the number pv systems with nan padding
+        n_gsp_per_example: the number gsp systems with nan padding
+        batch: if this example class is a batch or not
+    """
+
+    n_gsp_id = data[GSP_ID].shape[-1]
+    assert (
+        n_gsp_id == n_gsp_per_example
+    ), f"gsp_is is len {n_gsp_id}, but should be {n_gsp_per_example}"
+    assert data[GSP_YIELD].shape[-2:] == (
+        seq_len_30_minutes,
+        n_gsp_id,
+    ), f"gsp_yield is size {data[GSP_YIELD].shape}, but should be {(seq_len_30_minutes, n_gsp_id)}"
+    assert data[GSP_X_COORDS].shape[-1] == n_gsp_id
+    assert data[GSP_Y_COORDS].shape[-1] == n_gsp_id
+    assert data[GSP_DATETIME_INDEX].shape[-1] == seq_len_30_minutes
+
+    # check the GSP data is between 0 and 1
+    assert (
+        np.nanmax(data[GSP_YIELD]) <= 1.0
+    ), f"Maximum GSP value is {np.nanmax(data[GSP_YIELD])} but it should be <= 1"
+    assert (
+        np.nanmin(data[GSP_YIELD]) >= 0.0
+    ), f"Maximum GSP value is {np.nanmin(data[GSP_YIELD])} but it should be >= 0"
+
+    if OBJECT_AT_CENTER in data.keys():
+        assert data[OBJECT_AT_CENTER] == "gsp"
+
+    if not batch:
+        # add an extract dimension so that its similar to batch data
```

Comment:
Two tiny typos (sorry for being such a pedant... and sorry if I've misunderstood!)

* extract -> extra
* its -> it's

Reply:
you are definitely right! my bad